### DISCUSSION TITLE
CONFIGURE_TOTP Via API enpoint

### DISCUSSION THREAD

**Discussion by demetrius-powers** [keycloak/keycloak#28380] [Q&A] [answered]
Can I use this endpoint or is there a endpoint where I can dynamically toggle on OTP 

admin/realms/Histosuite/authentication/required-actions/CONFIGURE_TOTP

Looks like my admin UI uses the above but when I try it from postman I get 403 forbidden

**Comment by justin-tay** [comment, marked as answer]
That looks like the correct endpoint.

`PUT` `http://localhost:8080/admin/realms/Histosuite/authentication/required-actions/CONFIGURE_TOTP`

You might want to check if the `Authorization` `Bearer` is correct and hasn't expired yet.

You will need a user on your `Histosuite` realm with the `realm-management` client `manage-realm` role.

Or you will need a user on the `master` realm with the `Histosuite-realm` client `manage-realm` role.

Or you could just use the default admin user

```shell
curl \
  -d "client_id=admin-cli" \
  -d "username=admin" \
  -d "password=password" \
  -d "grant_type=password" \
  "http://localhost:8080/realms/master/protocol/openid-connect/token"
```
